pkg/virt-controller/watch/vmi.go

```diff
@@ -2134,3 +2152,78 @@ func (c *VMIController) handleDynamicInterfaceRequests(vmi *virtv1.VirtualMachin
 
 	return nil
 }
+
+func (c *VMIController) updateInterfaceStatus(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
+	originalVMIStatus := vmi.Status.DeepCopy().Interfaces
+	indexedVMIStatus := indexIfaceStatus(originalVMIStatus)
+
+	var newStatus []virtv1.VirtualMachineInstanceNetworkInterface
+	ifaceNamingScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
+	for _, network := range vmi.Spec.Networks {
+		status := virtv1.VirtualMachineInstanceNetworkInterface{Name: network.Name}
+		if _, ok := indexedVMIStatus[network.Name]; ok {
+			// Already have the status, modify if needed
+			status = indexedVMIStatus[network.Name]
+		}
+
+		podIfaceName, wasFound := ifaceNamingScheme[network.Name]
+		if !wasFound {
+			return fmt.Errorf("could not find the pod interface name for network [%s]", network.Name)
+		}
+
+		indexedPodIfaces := nonDefaultMultusNetworksIndexedByIfaceName(pod)
+		if _, found := indexedPodIfaces[podIfaceName]; found {
+			status.PodConfigDone = true
```

Comment:
Found some history about why it is bad to update the vmi interface status from both virt-controller and virt-handler. Not that I have any other suggestion - https://github.com/kubevirt/kubevirt/pull/2963#discussion_r426651733